ivy/functional/frontends/torch/blas_and_lapack_ops.py

```diff
@@ -130,6 +130,46 @@ def logdet(input):
     return ivy.det(input).log()
 
 
+@to_ivy_arrays_and_back
+def lu_unpack(lu_data, lu_pivots, unpack_data=True, unpack_pivots=True):
+    if lu_data.ndim < 2:
+        raise ValueError(
+            "The shape of x should be (*, M, N), but received ndim is"
+            f" [{lu_data.ndim} < 2]"
+        )
+    if lu_pivots.ndim < 1:
+        raise ValueError(
+            "The shape of Pivots should be (*, K), but received ndim is"
+            f" [{lu_pivots.ndim} < 1]"
+        )
+    m, n = lu_data.shape[-2:]
+    k = min(m, n)
+    if unpack_pivots:
+        permute = ivy.arange(m)
+        for i in range(lu_pivots.shape[-1]):
+            if i != lu_pivots[i] - 1:
+                permute[i], permute[lu_pivots[i] - 1] = (
+                    permute[lu_pivots[i] - 1],
+                    permute[i],
+                )
```

Comment:
Can you find a vectorized version of this code? I know it's `O(K)` but I think if there is a vectorized version would be better.

Reply:
Hey!  I tried vectorizing the above loop using this:
```
permute, indices = ivy.arange(m), ivy.arange(lu_pivots.shape[-1])
mask = ivy.where(pivots != lu_pivots - 1, True, False)
permute[indices], permute[lu_pivots[mask] - 1] = permute[lu_pivots[mask] - 1], permute[indices[mask]]
```
However, there are some edge cases where the pivot indices are repeated and in that case, permuting sequentially gives exact permutation matrix ordering.